src/core/gui/PageView.h

```diff
@@ -39,8 +41,9 @@ class Element;
 class PositionInputData;
 class Range;
 class TexImage;
+class OverlayBase;
 
-class XojPageView: public Redrawable, public PageListener {
+class XojPageView: public Redrawable, public PageListener, public xoj::view::Repaintable {
```

Comment:
We already have Redrawable, why we need an additional Repaintable.

Reply:
Ok, how about renaming `Redrawable` -> `LegacyRedrawable` and `Repaintable` -> `Redrawable` ? :D

Reply:
At least the name wouldn't be "stupid" and the other name states, that the type is about to replaced some time.